mat_nn_hip.hpp

```diff
@@ -101,19 +105,25 @@ double su3_mat_nn(std::vector<site> &a, std::vector<su3_matrix> &b, std::vector<
 
   // benchmark loop
   auto tstart = Clock::now();
+  tprofiling = tstart;
+
   for (int iters=0; iters<iterations+warmups; ++iters) {
     if (iters == warmups) {
       hipDeviceSynchronize();
       tstart = Clock::now();
-	  }
+      tprofiling = tstart;
+    }
     hipLaunchKernelGGL(k_mat_nn, dim3(blocksPerGrid), dim3(threadsPerBlock), 0, 0, d_a, d_b, d_c, total_sites);
   }
   hipDeviceSynchronize();
+  profile->kernel_time = (std::chrono::duration_cast<std::chrono::microseconds>(Clock::now()-tprofiling).count())/1.0e6;
```

Comment:
If I'm reading the code properly, `tprofiling` and `tstart` will always be the same, so I'm not sure why the timer is being stopped two separate times in immediate succession.

Reply:
I think you're right, In the beginning I think there was some more going on so I had separate timers just in case, but they can probably just be combined into one